### PR TITLE
Add gif.js w/ git auto-update via single package.json

### LINKED ISSUE #10040: [Request] Add gif.js
**Library name:** gif.js
**Git repository url:** https://github.com/jnordberg/gif.js
**npm package url(optional):** 
**License(s):**
**Official homepage:**
**Wanna say something? Leave message here:**

=====================
Notes from cdnjs maintainer:
You are welcome to add a library via sending pull request,
it'll be faster then just opening a request issue,
and please don't forget to read the guidelines for contributing, thanks!!

### PR DESCRIPTION
**Pull request by sashberd**
close #10040, cc @jnordberg

Pull request for issue: #10040 
Related issue(s): # #

Checklist for **Pull request** or **lib adding request issue** follows the conventions.

Note that if you are using a distribution purpose repository/package, please also provide the url and other related info like popularity of the source code repo/package.

# Profile of the lib
 * Git repository (required): https://github.com/jnordberg/gif.js
 * Official website (optional, not the repository): http://jnordberg.github.io/gif.js/
 * NPM package url (optional):
 * License and its reference: MIT
 * GitHub / Bitbucket popularity (required):
   - Count of watchers: 78
   - Count of stars: 2321
   - Count of forks: 231
 * NPM download stats (optional):
   - Downloads in the last day:
   - Downloads in the last week:
   - Downloads in the last month:

# Essential checklist
 * [ ] I'm the author of this library
   * [X ] I would like to add link to the page of this library on CDNJS on website / readme
 * [ X] This lib was not found on cdnjs repo
 * [ X] No already exist / duplicated issue and PR
 * [ X] The lib has notable popularity
   * [X ] More than 200 [Stars / Watchers / Forks] on [GitHub / Bitbucket]
   * [X ] More than 800 downloads stats per month on npm registry
 * [ ] Project has public repository on famous online hosting platform (or been hosted on npm)

# Auto-update checklist
 * [ X] Has valid tags for each versions (for git auto-update)
 * [ ] Auto-update setup
 * [ ] Auto-update target/source is valid.
 * [ ] Auto-update filemap is correct.

# Git commit checklist
 * [ X] The first line of commit message is less then 50 chars, be clean and clear, easy to understand.
 * [ X] The parent of the commit(s) in the PR is not old than 3 days.
 * [X ] Pull request is sending from a non-master branch with meaningful name.
 * [ X] Separate unrelated changes into different commits.
 * [X ] Use rebase to squash/fixup dummy/unnecessary commits into only one commit.
 * [ X] Close corresponding issue in commit message
 * [X ] Mention related issue(s), people in commit message, comment.

### CODE REVIEW

File: ajax/libs/gif.js/package.json [hunk 1]
```diff
@@ -0,0 +1,29 @@
+{
+  "name": "gif.js",
+  "description": "JavaScript GIF encoding library",
+  "filename": "gif.min.js",
+  "author": "Johan Nordberg <code@johan-nordberg.com>",
+  "homepage": "http://jnordberg.github.io/gif.js/",
```
Comment (sufuf3) [inline ajax/libs/gif.js/package.json]: Let's use `https`.
https://jnordberg.github.io/gif.js/
It's safety for browsing.
Thank you.

File: ajax/libs/gif.js/package.json [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+    "encoder"
+  ],
+  "autoupdate": {
+    "source": "git",
+    "target": "git://github.com/jnordberg/gif.js.git",
+    "fileMap": [
+      {
+        "basePath": "dist",
+        "files": [
+          "gif*"
```
Comment (sufuf3) [inline ajax/libs/gif.js/package.json]: Let's use the pattern `**/*`
As usual, we use `**/*` in `dist` as the basePath.
Thank you.

File: ajax/libs/gif.js/package.json [hunk 3]
```diff
@@ -0,0 +1,29 @@
+{
+  "name": "gif.js",
+  "description": "JavaScript GIF encoding library",
+  "filename": "gif.min.js",
```
Comment (sufuf3) [inline ajax/libs/gif.js/package.json]: Let's use `gif.js`
Because `gif.js` is minified file of this lib.
Thank you.

**Comment by sufuf3** [comment]
@cdnjs/active-contributor @cdnjs/intern3 @cdnjs/library-reviewer 
Please help review this PR, Thank you.

**Comment by maruilian11** [comment]
Helped rebase with the latest master.

**Comment by maruilian11** [comment]
@PeterDaveHello Please take a look. Thanks.

**Pull request merged**